src/rdkafka_cgrp.c

```diff
@@ -2650,8 +2667,14 @@ static rd_kafka_op_res_t
 rd_kafka_cgrp_consumer_handle_Metadata_op(rd_kafka_t *rk,
                                           rd_kafka_q_t *rkq,
                                           rd_kafka_op_t *rko) {
-        int i, j, found = 0;
+        /*
+         * FIXME: Using next_target_assignment is not correct as other heartbeat
+         * call can change it.
+         */
```

Comment:
It cannot happen because the responses are received and processed in order and the Metadata response will the always processed before the next HB response.

Reply:
As discussed over slack that this can happen. We will try to send metadata request to the coordinator in this case in future. Right now we should assess what can happen in this case.

Reply:
In the worst case, all the metadata for the next_target_assignment won't be present and in that case next_target_assignment is not freed. This will eventually follow partial ack flow. So we are good here.